src/Plugins/Loader/AssemblyLoadContextBuilder.cs

```diff
@@ -148,6 +148,9 @@ public AssemblyLoadContextBuilder PreferDefaultLoadContext(bool preferDefaultLoa
         /// <returns>The builder.</returns>
         public AssemblyLoadContextBuilder AddManagedLibrary(ManagedLibrary library)
         {
+            if (_managedLibraries.ContainsKey(library.Name.Name))
```

Comment:
I don't want to put this change in. The existing behavior is by design - adding multiple versions of the same library is not supported, and this usually means the builder is being called incorrectly.

I would be okay adding a different method, though, such as `bool TryManagedLibrary(ManagedLibrary library)`

Reply:
That would also work, but my problem is that it is not me, who is calling this method, but the `DependencyContextExtensions.AddDependencyContext`.

My situation is bit more complex than ususal, because I use your lib to load and analyze test projects of any arbitrary users, who are not explicitly implementing any new plugin. My general goal is that whatever a test runner can load I should also be able to load. And it looks like (see stack trace linked in [that issue](https://github.com/specsolutions/deveroom-visualstudio/issues/32)), sometimes there are duplicated in the deps file, but the test runners seem to be handle that. 

Maybe because my situation is special, the better solution would be to make this method virtual and make sure I can override it with an altered implementation. I will play with this and come back.